ansible_base/authentication/authenticator_plugins/local.py

```diff
@@ -54,6 +62,11 @@ def authenticate(self, request, username=None, password=None, **kwargs):
 
         user = super().authenticate(request, username, password, **kwargs)
 
+        if not user and not (request and request.path.startswith('/api/controller/')) and self.is_a_partially_migrated_controller_user(username, password):
```

Comment:
Not sure about this. I'm assuming you are doing this so you can make the request to `/api/controller/v2/me/`. But if you go to `https://<host>/api/controller/v2/instances/` and click login you will be taken to `https://34.207.163.197/api/controller/login` which actually looks like the gateway page and you can log in with gateway creds. So I think having this that open might prevent that from working? Its kind of a nice case that you are logging in for the first time to the API via a controller endpoint.

Reply:
Also, this portion of DAB is made to be used generically and this would prevent that from happening (making it very AAP specific). It might be better to have some kind of variable that would kick this logic off.

Reply:
I updated to instead ensure the path starts with /api/controller/v2/me/ to consider this

Reply:
I updated further to only check on `/api/gateway/v1/login/`

Reply:
Updated even further to be - 
```
and request.path.startswith(reverse('login', current_app='aap_gateway_api'))
```
Based on @chrismeyersfsu suggestion